### PR TITLE
DDF-4639 Update strategy for UI extensions

### PR DESCRIPTION
**Pull request by andrewkfiedler**
#### What does this PR do?
Updates strategy for UI extensions to leverage javascript / typescript.

This also involved pulling out 3 sub packages, now located here:
https://github.com/connexta/ace
https://github.com/connexta/eslint-plugin
https://github.com/connexta/icons

Also fixes an issue in https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/component/filter-builder/filter-serialization.js#L59 that was causing near filters to fail.

#### Who is reviewing it? 
@Bdthomson 
@djblue 
@adimka 
@rzwiefel 

#### How should this be tested?
Verify that the navigator and the routes work as expected.  Also, if you have access to a downstream project.  Verify that these changes don't break things there.  I have PRs in respective downstream projects to make them compatible with this PR.

#### What are the relevant tickets?
Fixes: #4639

### CODE REVIEW

File: ui/.eslintrc.json [hunk 1]
```diff
@@ -1,16 +0,0 @@
...
-        "sourceType": "module",
-        "ecmaFeatures": {
-            "jsx": true
-        }
-    },
-    "plugins": [
-        "codice-linter"
-    ],
-    "rules": {
-        "codice-linter/no-absolute-urls": 2
```
Comment (Bdthomson) [inline ui/.eslintrc.json]: Where does this live now?

File: ui/packages/catalog-ui-search/package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
   "scripts": {
     "fmt": "ace format -w",
     "prestart": "ace disable-idp",
-    "start": "ace start",
+    "start": "node --max_old_space_size=8192 ../../node_modules/@connexta/ace/bin.js start",
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/package.json]: Why do you need to give node this parameter for space size?

File: ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigator/navigator.presentation.tsx [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+
+export const Root = styled<Props, 'div'>('div')`
+  height: 100%;
+  width: 100%;
+  overflow: auto;
+`
+
+export const WorkspacesSave = styled<
+  {
+    isSaved: Props['isSaved']
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigator/navigator.presentation.tsx]: Why not just make this `styled<Props, 'div'....`?

File: ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigator/navigator.presentation.tsx [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+`
+
+export const WorkspacesIndicator = styled.div`
+  padding-left: ${props => props.theme.minimumSpacing};
+  display: inline-block;
+`
+
+export const SourcesIndicator = styled<
+  {
+    hasUnavailableSources: Props['hasUnavailableSources']
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigator/navigator.presentation.tsx]: Same here

File: ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigator/navigator.presentation.tsx [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+      {children}
+    </LinkBase>
+  )
+}
+
+export const ProductLink = ({ navigateToRoute, branding, product }: Props) => {
+  return (
+    <Link
+      onClick={() => {
+        navigateToRoute('workspaces')
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigator/navigator.presentation.tsx]: Why do `ProductLink` and `WorkspaceLink` both navigate to workspaces?

File: ui/packages/catalog-ui-search/src/main/webapp/js/Entry.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/**
+ * ApplicationSetup needs to happen first.
+ * This ensures styles are applied correctly,
+ * because some of our styles have the same specificity as vendor
+ * styles.
+ */
+;(function verifyFirstImport() {
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/src/main/webapp/js/Entry.tsx]: Would you rather add a `@ts-ignore` here?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx [hunk 7]
```diff
@@ -17,16 +17,16 @@ class DocumentListener extends React.Component {
   }
 }
 
-class Menu extends React.Component {
-  constructor(props) {
+export class Menu extends React.Component<any, any> {
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx]: ✏️ Could export `hot(module)` these

File: ui/.eslintrc.json (hunk 1 above)
Comment (andrewkfiedler) [reply ui/.eslintrc.json]: https://github.com/connexta/eslint-plugin

File: ui/packages/catalog-ui-search/package.json (hunk 2 above)
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/package.json]: We were running into out of memory issues.

File: ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigator/navigator.presentation.tsx (hunk 3 above)
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigator/navigator.presentation.tsx]: I'd rather pass in exactly what we need in this case.

File: ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigator/navigator.presentation.tsx (hunk 4 above)
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigator/navigator.presentation.tsx]: Similar to above, I'd rather pass in exactly what we need in this case.

**Comment by andrewkfiedler** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js [hunk 8]
```diff
@@ -107,7 +108,36 @@ function comparatorToCQL() {
...
+          <span className="fa fa-minus" />
+        </button>
+        <div
+          className="filter-attribute"
+          data-help="Property to compare against."
+        />
+        <div
+          className="filter-comparator"
+          data-help="How to compare the value for this property against
+the provided value."
```
Comment (rzwiefel) [inline ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js]: is this line broken?

Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js]: Nope

**Comment by andrewkfiedler** [comment]
build now

**Comment by andrewkfiedler** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/index.tsx [hunk 9]
```diff
@@ -0,0 +1,3 @@
+import Entry from './js/Entry'
+
+Entry()
```
Comment (bakejeyner) [inline ui/packages/catalog-ui-search/src/main/webapp/index.tsx]: Is `export` not needed here?

Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/index.tsx]: It's unnecessary because this we don't expect anyone to pull in anything from this file.

**Comment by Bdthomson** [comment]
✅ Hero Successful
- Verified you could get to all routes

**Pull request merged**